data-serving/data-service/src/controllers/preprocessor.ts

```diff
@@ -113,6 +113,46 @@ export const setBatchRevisionMetadata = async (
                     c.caseReference?.sourceEntryId,
             ) || createNewMetadata(curatorEmail);
     });
+    // Clean up the additional metadata that falls outside the `case` entity.
+    delete request.body.curator;
+
+    next();
+};
+
+export const setBatchUpdateRevisionMetadata = async (
+    request: Request,
+    response: Response,
+    next: NextFunction,
+) => {
+    const curatorEmail = request.body.curator.email;
+
+    const existingCases = await Case.find({
+        _id: {
+            $in: request.body.cases.map((c: any) => c._id),
+        },
+    })
+        .select({
+            _id: 1,
+            caseReference: 1,
+            revisionMetadata: 1,
+        })
+        .exec();
+
+    const metadataMap = new Map(
+        existingCases
+            .filter((c) => c && c.caseReference)
+            .map((c) => [
+                c.caseReference.sourceId + ':' + c.caseReference.sourceEntryId,
```

Comment:
For this method, we aren't guaranteed to have unique sourceId/sourceEntryId combinations. Do we need the key of this metadataMap to be unique?

Reply:
Yes, good call. Switched to using ids instead since that is required in the body.